### PR TITLE
compressed payload

### PR DESCRIPTION
**Pull request by AelMalinka**

### CODE REVIEW

File: test/large-incident.js [hunk 1]
```diff
@@ -0,0 +1,139 @@
+import fs from 'fs';
+import moment from 'moment';
+import sizeof from 'object-sizeof';
+import zlib from 'zlib';
+import async from 'async';
+import request from 'requestretry';
+import aws4 from 'aws4';
+
+describe('API', () => {
```
Comment (pfrollo) [inline test/large-incident.js:9]: You could add a single assert statement for each test below.... assert(body.size()).less.then(100kb) <- pseudo code

File: src/index.js [hunk 2]
```diff
@@ -69,7 +70,9 @@ export default class StatEngine {
         request(req, (err, response) => {
           if (err) {
```
Comment (pfrollo) [inline src/index.js:71]: Please log the error here to make troubleshooting easier in the future

File: src/index.js [hunk 3]
```diff
@@ -69,7 +70,9 @@ export default class StatEngine {
...
             return postCb(err);
-          } else if (response.statusCode !== 204) {
+          }
+
+          if (response.statusCode !== 204) {
             return postCb(new Error(`Unexpected response: ${response.statusCode}`));
```
Comment (pfrollo) [inline src/index.js:76]: Please log the error here to make troubleshooting easier in the future

**Review by pfrollo — APPROVED** [review]

File: src/index.js (hunk 3 above)
Comment (AelMalinka) [reply src/index.js:78]: There isn't an error here and the response code is logged? I updated to log the above; did you want me to log more of the response object here?

Comment (pfrollo) [reply src/index.js:78]: It is fine. Thanks. Not sure if the extra logging of response.statusCode right here would gain us anything.

**Pull request merged**